File: solutions/python/graphs/find-the-town-judge.py
# LeetCode 997 — Find the Town Judge (Easy)
# Category: Graphs · Approach: Score Counting
# Time: O(n + e) | Space: O(n)
# Source: https://leetcode.com/problems/find-the-town-judge/

def find_judge(n: int, trust: list[list[int]]) -> int:
    score = [0] * (n + 1)
    for a, b in trust:
        score[a] -= 1
        score[b] += 1
    for i in range(1, n + 1):
        if score[i] == n - 1:
            return i
    return -1

File: solutions/typescript/graphs/find-the-town-judge.ts
// LeetCode 997 — Find the Town Judge (Easy)
// Category: Graphs · Approach: Score Counting
// Time: O(n + e) | Space: O(n)
// Source: https://leetcode.com/problems/find-the-town-judge/

function findJudge(n: number, trust: number[][]): number {
  const score = new Array(n + 1).fill(0);
  for (const [a, b] of trust) {
    score[a]--;
    score[b]++;
  }
  for (let i = 1; i <= n; i++) {
    if (score[i] === n - 1) return i;
  }
  return -1;
}
